software/image_processor.py

```diff
@@ -84,32 +85,78 @@ def start(self):
-    def get_lines(self, image):
+    def get_lines(self, image, fragmented):
         img = cv2.cvtColor(image,cv2.COLOR_BGR2GRAY)
         # 30:400 proved to work
-        gr_img = img[30:320]
+        gr_img = img[0:320]
         krn = 1 # kernel size for gauss
-        blur_img = cv2.GaussianBlur(gr_img, (krn, krn), 0)
+        #blur_img = cv2.GaussianBlur(gr_img, (krn, krn), 0)
         
+        
+
+        frag_sx, frag_sy = np.shape(fragmented)
+        fragmentedwhite = np.zeros((frag_sx, frag_sy))
+        fragmentedblack = np.zeros((frag_sx, frag_sy))
+        #print(f"tere hommikust: {frag_sx} {frag_sy}")
+#        comb_img = np.zeros((frag_sx, frag_sy))
+
+        
+        #fragmentedblack[fragmentedblack == 6] = 0
+        fragmentedblack[fragmented == 6] = 1
+
+        #fragmentedwhite[fragmentedwhite == 5] = 0
+        fragmentedwhite[fragmented == 5] = 1
+        
+        
+        openkernel = np.ones((3,3), np.uint8)
+        lowdilatekernel = np.ones((3,3), np.uint8)
+#        erodekernel = np.ones((3,3), np.uint8)
 
-        low = 80
-        high = 120
+        # d notation - dilated version - used for detecting lines
+#        white_img_d = cv2.dilate(white_img, dilatekernel)
+#        black_img_d = cv2.dilate(black_img, dilatekernel)
+#        white_img_d = cv2.erode(white_img_d, erodekernel)
+#        black_img_d = cv2.erode(black_img_d, erodekernel)
+
+        # without d notation - a bit dilated images, used for checking colours.
+        detectionblack = cv2.morphologyEx(fragmentedblack, cv2.MORPH_CLOSE, openkernel)
+
+        fragmentedwhite = cv2.dilate(fragmentedwhite, lowdilatekernel)
+        fragmentedblack = cv2.dilate(fragmentedblack, lowdilatekernel)
+
+        
+
+        fragmentedwhite = fragmentedwhite.astype(np.uint8) * 255
+        fragmentedblack = fragmentedblack.astype(np.uint8) * 255
+        detectionblack = detectionblack.astype(np.uint8) * 255
+
+#        comb_img = np.logical_and(white_img_d, black_img_d)
+#        comb_img = comb_img.astype(np.uint8) * 255
+        
+        #print("bit " + str(np.bitwise_and(6, 5)))
+        #print("log " + str(np.logical_and(6, 5)))
+        #print(comb_img)
 
-        ret, thresh = cv2.threshold(blur_img, low, high, cv2.THRESH_BINARY_INV)
+        low = 60
+        high = 100
+
+        ret, thresh = cv2.threshold(gr_img, low, high, cv2.THRESH_BINARY_INV)
 
 
         # different line detection parameters
         #scv2.imshow('hallo', thresh)
         low_thr = 50
-        high_thr = 150
-        edges = cv2.Canny(thresh, low_thr, high_thr)
+        high_thr = 160
+        edges = cv2.Canny(detectionblack, low_thr, high_thr)
         rho = 1
         theta = np.pi / 180 * 1
-        threshold = 50
-        minline = 100
-        maxgap = 40
+        threshold = 30
+        minline = 60
+        maxgap = 30
+
+        
 
-        cropped = image[30:320]
+        cropped = image[0:320]
 
         copyimg = np.copy(cropped) * 0
```

Comment:
FYI there is already debug frame provided in the code. Lines 56 and 57 (debug_frame).

Reply:
This was not always used for debug, I would keep it there for now in case we need to rewrite it closer to its original form.